src/systemcmds/tests/test_versioning.cpp

```diff
@@ -0,0 +1,117 @@
+#include <unit_test.h>
+#include <version/version.h>
+
+class VersioningTest : public UnitTest
+{
+public:
+	virtual bool run_tests();
+
+private:
+	bool _is_correct_version_tag(const char *version_tag, uint32_t result_goal);
+	bool _is_correct_version_tag_vendor(const char *version_tag, uint32_t result_goal);
+
+	bool _test_flight_version();
+	bool _test_vendor_version();
+};
+
+bool VersioningTest::_is_correct_version_tag(const char *version_tag, uint32_t result_goal)
+{
+	uint32_t result = version_tag_to_number(version_tag);
+
+	if (result == result_goal) {
+		return true;
+
+	} else {
+		PX4_ERR("Wrong version: tag: %s, got: 0x%x, expected: 0x%x", version_tag, result, result_goal);
+		return false;
+	}
+}
+
+bool VersioningTest::_is_correct_version_tag_vendor(const char *version_tag, uint32_t result_goal)
+{
+	uint32_t result = version_tag_to_vendor_version_number(version_tag);
+
+	if (result == result_goal) {
+		return true;
+
+	} else {
+		PX4_ERR("Wrong version: tag: %s, got: 0x%x, expected: 0x%x", version_tag, result, result_goal);
+		return false;
+	}
+}
+
+bool VersioningTest::run_tests()
+{
+	ut_run_test(_test_flight_version);
+	ut_run_test(_test_vendor_version);
+
+	return (_tests_failed == 0);
+}
+
+bool VersioningTest::_test_flight_version()
+{
+	ut_assert_true(_is_correct_version_tag("v11.45.99-1.2.3", 0xB2D63ff));
```

Comment:
I think it would be better to have a combined test function, like `_is_correct_version(string tag, int expected_px4_version, int expected_vendor_version)`, to avoid duplicating the list of strings.

Reply:
Good input, thanks! Done cf2fdb5